activerecord/lib/active_record/reflection.rb

```diff
@@ -618,6 +622,30 @@ def derive_join_table
         def primary_key(klass)
           klass.primary_key || raise(UnknownPrimaryKey.new(klass))
         end
+
+        # Finds inverse reflection from +klass+ with properties equal to options with provided keys.
+        def find_inverse_reflection_by_options keys
+          result = klass.reflections.find do |name, refl|
+            # We skip polymorphic and associations with :as option because we can't determine their classes.
+            !refl.polymorphic? && !refl.options.key?(:as) && refl.klass == active_record && keys.all? do |key|
```

Comment:
This seems to care quite a bit about the internals of other associations. Can you give a name to what it's looking for?